### ISSUE TITLE
Shops selling default items when using a custom item list

### ISSUE THREAD

**Issue by Schattenmensch** [az64/mm-rando#14]
I used a custom item list to shuffle everything with glitchless logic except the two swamp archery rewards (because fuck doing that one with a keyboard ^^) and the shopkeepers where selling the same items as vanilla/ refusing to sell an item if i didn't match the vanilla requirements to buy it while according to the spoiler log they should be selling Woodfall dungeon items and a Lens of Truth.
Tried again with slightly different settings (everything except the swamp archery rewards and the bottle pickup randomisation) and got the same result, while generating a similar rom using the built in random elements has the shopkeepers working as intended.

**Comment by pkmnfrk** [comment]
Just to clarify, in the cases where items did not have special requirements, were they still selling the defaults? Like, if you try to buy a stick, did you get a stick or the actual item?

In my observance, without using custom items, some items are still locked behind requirements. Eg, Koume will not sell the blue potion by default (presumably until you provide a mushroom), even if it was something else. On the other hand, I was able to buy the bomb and bombchu items from the bomb shop without a bomb bag, so who knows.

**Comment by Schattenmensch** [comment]
Yeah, just to clarify: Every item i could buy while using the custom item list was the default one despite the spoiler log claiming it should be something else. 
The restrictions I mentioned are stuff like buying the shield at the trading post if you've already got a shield, which does work in a randomized rom without the custom item list but doesn't work with the custom item list and in vanilla.  (presumably because it would have been the vanilla item)
So basically the traders (I tried Trading Post, Bomb Shop and Koume) where acting entirely as they do in vanilla.
